src/app/storage/duck/sagas.ts

```diff
@@ -170,18 +193,36 @@ function* addStorageRequest(action) {
   // If any of the objects actually fail creation, we need to rollback the others
   // so the storage is created, or fails atomically
   try {
-    const storageAddResults = yield Q.allSettled([
-      client.create(secretResource, storageSecret),
-      client.create(migStorageResource, migStorage),
-    ]);
+    const storageAddResults = [];
+    const storageSecretAddResult = yield client.create(secretResource, storageSecret);
+
+    if (storageSecretAddResult.status === 201) {
+      storageAddResults.push(storageSecretAddResult);
+
+      Object.assign(migStorage.spec.backupStorageConfig.credsSecretRef, {
+        name: storageSecretAddResult.data.metadata.name,
+        namespace: storageSecretAddResult.data.metadata.namespace,
+      });
+
+      Object.assign(migStorage.spec.volumeSnapshotConfig.credsSecretRef, {
+        name: storageSecretAddResult.data.metadata.name,
+        namespace: storageSecretAddResult.data.metadata.namespace,
+      });
+
+      const storageAddResult = yield client.create(migStorageResource, migStorage);
+
+      if (storageAddResult.status === 201) {
```

Comment:
I don't think this should only get pushed to the results array if it was successfully created. We want a failed result to trigger the rollback?

Reply:
@eriknelson This will work. Notice changes on line 224 where I have changed the logic to look for actual response status code, instead of the promise' fulfilled or rejected status

Reply:
@eriknelson So I assumed when we create a cluster, it either gets created or it doesn't. I did not think that there's a half way outcome. According to that logic, I thought it would be redundant to add it to rollback objects.